swap/peer.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package swap
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	contract "github.com/ethersphere/swarm/contracts/swap"
+	"github.com/ethersphere/swarm/p2p/protocols"
+)
+
+// ErrDontOwe indictates that no balance is actially owned
+var ErrDontOwe = errors.New("no negative balance")
+
+// Peer is a devp2p peer for the Swap protocol
+type Peer struct {
+	*protocols.Peer
+	swap               *Swap
+	backend            contract.Backend
+	beneficiary        common.Address
+	contractAddress    common.Address
+	lastReceivedCheque *Cheque
+}
+
+// NewPeer creates a new swap Peer instance
+func NewPeer(p *protocols.Peer, s *Swap, backend contract.Backend, beneficiary common.Address, contractAddress common.Address) *Peer {
```

Comment:
this contructor should load and set balance and sent and received cheques from statestore, see below

Reply:
ditto

Reply:
relevant issue: #1604.